chat/client-api.md

```diff
@@ -0,0 +1,62 @@
+# Client API
+
+```typescript
+abstract class Client {
+  // ---------- Methods ----------------------------------------------- //
+
+  // initializes the client with persisted storage and a network connection
+  public abstract init(): Promise<void>;
+
+  // register a blockchain account with a public key / returns the public key
+  public abstract register(params: {
+    account: string;
+  }): Promise<string>;
+
+  // queries the default keyserver with a blockchain account / returns the public key
+  public abstract resolve(params: {
+    account: string;
+  }): Promise<string>;
+
+  // sends a chat invite with opening message / returns an invite id
+  public abstract invite(params: {
+    peerPublicKey: string;
+    invite: Invite;
+  }): Promise<number>;
```

Comment:
we are inviting with a peerPublicKey - that method will work for all types of discoverability but we probably should add a label or an Account as an argument so inviter can associate something human readable with it's peer.

Reply:
The Invite struct already includes the account as a parameter

Reply:
true, account of an inviter but I mean invitee account on inviter side so inviter can later associate that pubKey with someone.